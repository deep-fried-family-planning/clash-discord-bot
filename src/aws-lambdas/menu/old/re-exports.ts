export {
    ApplicationCommandType as CMD,
    ApplicationCommandOptionType as CMD_OP,
    InteractionType as ITR,
    MessageFlags as MSG,
    WebhookType as WBH,
    TextInputStyle as CMP_T,
    // ButtonStyle as CMP_B,
} from '@discordjs/core/http-only';

import {Discord} from 'dfx';

export type CMP = Discord.ComponentType;
export const CMP = Discord.ComponentType;
export type CMP_B = Discord.ButtonStyle;
export const CMP_B = Discord.ButtonStyle;
