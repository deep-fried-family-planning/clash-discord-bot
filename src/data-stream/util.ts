import type {ServerClan} from '#src/data/index.ts';
import type {CreateScheduleCommandInput} from '@aws-sdk/client-scheduler';
import type {DeleteCommandInput, PutCommandInput, TransactWriteCommandInput, UpdateCommandInput} from '@aws-sdk/lib-dynamodb';
import type {ClanWar} from 'clashofclans.js';
import * as Data from 'effect/Data';

export type Terminal = Data.TaggedEnum<{
  Delete       : DeleteCommandInput;
  Put          : PutCommandInput;
  Update       : UpdateCommandInput;
  TransactWrite: TransactWriteCommandInput;
  Schedule     : CreateScheduleCommandInput;
}>;

export const Terminal = Data.taggedEnum<Terminal>();
