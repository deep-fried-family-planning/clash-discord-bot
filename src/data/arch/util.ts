import type {Discord} from 'dfx';
import * as Data from 'effect/Data';

export type RegistryCaller = {
  guild? : Discord.Guild;
  user?  : Discord.User;
  member?: Discord.GuildMember;
};

export type RegistrySuccess = {
  title      : string;
  description: string;
};

export class RegistryDefect extends Data.TaggedError('RegistryDefect')<{
  cause?: any;
}> {}

export class RegistryFailure extends Data.TaggedError('RegistryError')<{
  cause?: any;
}> {}

export class RegistryUserError extends Data.TaggedError('RegistryUserError')<{
  message?: string;
  cause?  : any;
}> {}

export class RegistryAdminError extends Data.TaggedError('RegistryAdminError')<{
  message?: string;
  cause?  : any;
}> {}
