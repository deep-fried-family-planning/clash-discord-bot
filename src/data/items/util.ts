import * as S from 'effect/Schema';

const ApiEmbedType = S.Enums({
  RICH       : 'rich',
  IMAGE      : 'image',
  VIDEO      : 'video',
  GIFV       : 'gifv',
  ARTICLE    : 'article',
  LINK       : 'link',
  POLL_RESULT: 'poll_result',
} as const);

const ApiEmbedProvider = S.Struct({
  name: S.optional(S.String),
  url : S.optional(S.String),
});

const ApiEmbedImage = S.Struct({
  url      : S.optional(S.String),
  proxy_url: S.optional(S.String),
  height   : S.optional(S.Number),
  width    : S.optional(S.Number),
});

export const ApiEmbed = S.Struct({
  type     : S.optional(ApiEmbedType),
  provider : S.optional(ApiEmbedProvider),
  image    : S.optional(ApiEmbedImage),
  thumbnail: S.optional(S.Struct({
    url      : S.optional(S.String),
    proxy_url: S.optional(S.String),
    height   : S.optional(S.Number),
    width    : S.optional(S.Number),
  })),
  video: S.optional(S.Struct({
    url      : S.optional(S.String),
    proxy_url: S.optional(S.String),
    height   : S.optional(S.Number),
    width    : S.optional(S.Number),
  })),
  author: S.optional(S.Struct({
    name          : S.String,
    url           : S.optional(S.String),
    icon_url      : S.optional(S.String),
    proxy_icon_url: S.optional(S.String),
  })),
  color      : S.optional(S.Number),
  title      : S.optional(S.String),
  url        : S.optional(S.String),
  description: S.optional(S.String),
  footer     : S.optional(S.Struct({
    text          : S.optional(S.String),
    icon_url      : S.optional(S.String),
    proxy_icon_url: S.optional(S.String),
  })),
  fields: S.optional(S.Array(S.Struct({
    name  : S.String,
    value : S.String,
    inline: S.optional(S.Boolean),
  }))),
  timestamp: S.optional(S.String),
});
