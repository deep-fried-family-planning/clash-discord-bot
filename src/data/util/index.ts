export enum ClanVerification {
  admin     = 0,
  elder     = 1,
  coleader  = 2,
  leader    = 3,
  developer = 4,
}

export enum PlayerVerification {
  none      = 0,
  admin     = 1,
  token     = 2,
  developer = 3,
}
