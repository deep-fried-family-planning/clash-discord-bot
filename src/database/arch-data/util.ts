import {E, S} from '#src/internal/pure/effect.ts';
import {decode, encode} from '@msgpack/msgpack';
import {DateTime, type Duration, pipe} from 'effect';
import {deflate, inflate} from 'pako';

export const Created = S.transformOrFail(
  S.DateTimeUtc,
  S.UndefinedOr(S.DateTimeUtcFromSelf),
  {
    decode: (dt) => E.succeed(dt),
    encode: (dt) => dt ? E.succeed(dt) : DateTime.now,
  },
);

export const Updated = S.transformOrFail(
  S.DateTimeUtc,
  S.UndefinedOr(S.DateTimeUtcFromSelf),
  {
    decode: (dt) => E.succeed(dt),
    encode: () => DateTime.now,
  },
);

export const TimeToLive = (duration: Duration.Duration) =>
  S.transformOrFail(
    S.DateTimeUtcFromNumber,
    S.UndefinedOr(S.DateTimeUtcFromSelf),
    {
      decode: (dt) => E.succeed(dt),
      encode: (dt) => dt
        ? E.succeed(dt)
        : pipe(
          DateTime.now,
          E.map(
            DateTime.addDuration(duration),
          ),
        ),
    },
  );

export const Decoded = S.transform(
  S.Undefined,
  S.Literal(true),
  {
    decode: () => true as const,
    encode: () => undefined,
  },
);

export const Upgraded = pipe(
  S.Union(S.Undefined, S.Boolean),
  S.transform(
    S.Union(S.Undefined, S.Boolean),
    {
      decode: (enc) => enc,
      encode: () => undefined,
    },
  ),
  S.optional,
);

export const SelectMetadata = <A, I, R>(value: S.Schema<A, I, R>) =>
  S.Struct({
    value      : value,
    label      : S.String,
    description: S.optional(S.String),
    default    : S.optional(S.Boolean),
    emoji      : S.optional(S.Struct({
      name: S.String,
    })),
  });

export const CompressedBinary = <A, I, R>(schema: S.Schema<A, I, R>) =>
  S.transform(S.String, schema, {
    decode: (enc) => {
      const buff = Buffer.from(enc, 'base64');
      const pako = inflate(buff);
      return decode(pako) as any;
    },
    encode: (dec) => {
      const pack = encode(dec);
      const pako = deflate(pack);
      return Buffer.from(pako).toString('base64');
    },
  });
