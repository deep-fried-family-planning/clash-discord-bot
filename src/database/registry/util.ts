import {Data} from 'effect';

export class RegistrationError extends Data.TaggedError('deepfryer/RegistrationError')<{
  issue: string;
}> {}

export class RegistrationDefect extends Data.TaggedError('deepfryer/RegistrationDefect')<{
  issue: string;
}> {}
