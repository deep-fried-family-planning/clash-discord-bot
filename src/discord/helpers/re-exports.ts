export {
    ApplicationCommandType as CMD,
    ApplicationCommandOptionType as CMD_OP,
    InteractionType as ITR,
    MessageFlags as MSG,
    WebhookType as WBH,
    ComponentType as CMP,
    TextInputStyle as CMP_T,
    ButtonStyle as CMP_B,
} from 'discord-api-types/v10';
