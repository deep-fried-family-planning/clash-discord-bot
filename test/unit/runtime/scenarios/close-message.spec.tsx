import {usePage} from '#src/disreact/index.ts';
import {E} from '#src/disreact/utils/re-exports.ts';
import {it} from '@effect/vitest';
import {makeTestRuntime} from 'test/unit/util.ts';



const CloseButton = () => {
  const page = usePage([]);
  return (<secondary
    custom_id={'CloseButton'}
    onclick={() => page.close()}
  />);
};

const MessageClose = () => <message>
  <embed>
    {'Hello World!'}
  </embed>
  <actions>
    <CloseButton/>
  </actions>
</message>;

const runtime = makeTestRuntime([MessageClose], false);

it.scoped('when closing', E.fn(function* () {
  const root = yield* runtime.synthesize(MessageClose);

  expect(root).toMatchSnapshot();

  const res = yield* runtime.respond({
    id            : '1236074574509117491',
    token         : 'respond1',
    application_id: 'app',
    user_id       : 'user',
    guild_id      : 'guild',
    message       : root,
    type          : 2,
    data          : {
      custom_id     : 'CloseButton',
      component_type: 2,
    },
  });

  expect(res).toMatchSnapshot();
  expect(runtime.createUpdate).not.toBeCalled();
  expect(runtime.deferEdit).not.toBeCalled();
  expect(runtime.deferUpdate).toBeCalled();
  expect(runtime.dismount).toBeCalled();
}));
