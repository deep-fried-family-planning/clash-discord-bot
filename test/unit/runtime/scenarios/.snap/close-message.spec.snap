// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when closing 1`] = `
{
  "components": [
    {
      "components": [
        {
          "custom_id": "CloseButton",
          "disabled": undefined,
          "emoji": undefined,
          "label": undefined,
          "style": 2,
          "type": 2,
        },
      ],
      "type": 1,
    },
  ],
  "content": undefined,
  "embeds": [
    {
      "author": undefined,
      "color": undefined,
      "description": "Hello World!",
      "fields": undefined,
      "footer": undefined,
      "image": {
        "url": "https://dffp.org/s/eJxrXpSZssY5J7841Te1uDgxPXVpQVF-QfGBZcUlicnZxU0ochNu6iFzrXKhtIFVYnJJZn5esZWhFViBU2lJSX6elcHEAwDcQifj",
      },
      "title": undefined,
      "url": undefined,
    },
  ],
  "flags": undefined,
}
`;

exports[`when closing 2`] = `null`;
